File: day4/c++/day4_test.cpp
#define CATCH_CONFIG_MAIN
#include "catch.hpp"
#include "./day4_lib.h"

TEST_CASE( "Int to vector works", "[lib]" ) {
    REQUIRE( int_to_vector(1234) == std::vector<int>{1,2,3,4} );
    REQUIRE( int_to_vector(158912) == std::vector<int>{1,5,8,9,1,2} );
}

TEST_CASE( "Valid password function works", "[simple]" ){
    REQUIRE( true  == valid_password(111111) );
    REQUIRE( false == valid_password(223450) );
    REQUIRE( false == valid_password(123789) );
    REQUIRE( true  == valid_password(556789) );
    REQUIRE( false == valid_password(554789) );
}

TEST_CASE( "Valid password 2 function works", "[complex]" ){
    REQUIRE( false == valid_password_2(111111) );
    REQUIRE( true  == valid_password_2(556789) );
    REQUIRE( false == valid_password_2(223450) );
    REQUIRE( false == valid_password_2(555789) );
    REQUIRE( true  == valid_password_2(112233) );
    REQUIRE( false == valid_password_2(123444) );
    REQUIRE( true  == valid_password_2(111122) );
    REQUIRE( true  == valid_password_2(112222) );
}

File: day04/ruby/day4_test.rb
require "test/unit"
require "./day4_lib.rb"

class LibTests < Test::Unit::TestCase
  def test_password_checker
    assert_equal true,  valid_password(111111)
    assert_equal false, valid_password(223450)
    assert_equal false, valid_password(123789)
    assert_equal true,  valid_password(556789)
    assert_equal false, valid_password(554789)
  end

  def test_password_checker_2
    assert_equal false, valid_password_2(111111)
    assert_equal true,  valid_password_2(556789)
    assert_equal false, valid_password_2(223450)
    assert_equal false, valid_password_2(555789)
    assert_equal true,  valid_password_2(112233)
    assert_equal false, valid_password_2(123444)
    assert_equal true,  valid_password_2(111122)
    assert_equal true,  valid_password_2(112222)
  end
end

